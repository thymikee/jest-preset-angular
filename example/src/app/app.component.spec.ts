import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { async, fakeAsync, tick, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { ConfigureFn, configureTests } from '@lib/testing';

import { AppComponent } from './app.component';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let component: AppComponent;

  beforeEach(
    async(() => {
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [AppComponent],
          imports: [NoopAnimationsModule],
          schemas: [NO_ERRORS_SCHEMA],
        });
      };

      configureTests(configure).then(testBed => {
        fixture = testBed.createComponent(AppComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
      });
    })
  );

  it(
    'should create the app',
    async(() => {
      const app = component;
      expect(app).toBeTruthy();
    })
  );

  it('snaps', () => {
    expect(fixture).toMatchSnapshot();
  });

  it(
    `should have as title 'app works!'`,
    async(() => {
      const app = component;
      expect(app.title).toEqual('app works!');
    })
  );

  it(
    'should render title in a h1 tag',
    async(() => {
      const compiled = fixture.debugElement.nativeElement;
      expect(compiled.querySelector('h1').textContent).toContain('app works!');
    })
  );

  it('looks async but is synchronous', <any>fakeAsync((): void => {
    let flag = false;
    setTimeout(() => {
      flag = true;
    }, 100);
    expect(flag).toBe(false);
    tick(50);
    expect(flag).toBe(false);
    tick(50);
    expect(flag).toBe(true);
  }));
});
