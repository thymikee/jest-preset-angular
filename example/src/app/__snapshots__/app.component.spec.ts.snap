// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent snaps 1`] = `
<app-root
  __ngContext__={[Function LRootView]}
  hasClass={[Function Boolean]}
  title={[Function String]}
  variableWithPrecedingDolar={[Function Number]}
>
  <h1
    class="ng-tns-c11-0"
  >
     app works! 
    <app-calc
      class="ng-tns-c11-0"
    />
    <span
      class="ng-tns-c11-0"
    >
      aaa $1234
    </span>
    <span
      class="ng-tns-c11-0 ng-star-inserted"
      style=""
    >
      ddd
    </span>
    
     \`test'chars""
  
  </h1>
</app-root>
`;
